tcp-server/src/main.rs

```diff
@@ -0,0 +1,116 @@
+use std::borrow::BorrowMut;
+use std::collections::vec_deque::Iter;
+use std::collections::VecDeque;
+use std::io;
+use std::io::{BufRead, BufReader, Read, Write};
+use std::net::{TcpListener, TcpStream};
+use std::ops::Deref;
+use std::sync::{Arc, Mutex};
+use std::thread;
+use std::thread::JoinHandle;
+use std::time::Duration;
+
+use redisish::{Command, parse};
+
+trait Mailbox {
+    /// TODO Is it fine that mutability is hidden? How can we do it some other way?
+    fn append(&self, email: String);
```

Comment:
I'd make it explicit by declaring `append(&mut self`

Reply:
In order to achieve that I will need a lock around it, right? The struct is used from multiple threads/tasks/functions.

What would be preferable in this case, a lock around the struct or a lock inside the struct?